start/lib/Service/PdoFleetStorage.php

```diff
@@ -0,0 +1,172 @@
+        $stmt = $this->pdo->prepare($query);
+        $stmt->execute();
+        $shipFleetArray = $stmt->fetchAll(\PDO::FETCH_ASSOC);
+
+        return $shipFleetArray;
+    }
+
+    public function findTeams(): array
+    {
+        $query = 'SELECT
+                f.team
+            FROM
+                fleets f
+            GROUP BY
+                f.team';
+
+        $statement = $this->pdo->prepare($query);
+        $statement->execute();
+
+        $results = $statement->fetchAll();
+
+        $teams = [];
+        foreach ($results as $result) {
+            $teams[] = $result['team'];
+        }
+
+        return $teams;
+    }
+
+    public function findFleetById($id): array
+    {
+        $query = 'SELECT * FROM fleets WHERE id = :id';
+
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':id',$id);
+        $stmt->execute();
+        $results = $stmt->fetch(\PDO::FETCH_ASSOC);
+
+        return $results;
+    }
+
+    public function findFleetShipsByFleet(Fleet $fleet): array
+    {
+        $query = 'SELECT * FROM ship_fleets WHERE fleet_id = :fleetId';
+
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':fleetId', $fleet->getId());
+        $stmt->execute();
+        $results = $stmt->fetchAll(\PDO::FETCH_ASSOC);
+
+        return $results;
+    }
+
+    public function fetchSingleFleetById($id): ?array
+    {
+        $query = 'SELECT fleets.id, ship.id ship_id, ship.name ship_name, fleets.name, fleets.team, sum(ship_fleets.quantity) quantity FROM fleets JOIN ship_fleets ON fleets.id = ship_fleets.fleet_id JOIN ship ON ship.id = ship_fleets.ship_id WHERE fleets.id = :id GROUP BY ship_name';
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':id', $id);
+        $stmt->execute();
+        $fleet = $stmt->fetchAll(\PDO::FETCH_ASSOC);
+
+        if ($fleet === []) {
+            return null;
+        }
+
+        return $fleet;
+    }
+
+    public function deleteFleet($id): void
+    {
+        $query = 'DELETE FROM ship_fleets WHERE fleet_id = :id';
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':id', $id);
+        $stmt->execute();
+
+        $query = 'DELETE FROM fleets WHERE id = :id';
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':id', $id);
+        $stmt->execute();
+    }
+
+    public function deleteShipFromFleet(ShipFleet $fleetShip): void
+    {
+        $query = 'DELETE FROM ship_fleets WHERE fleet_id = :fleetId AND ship_id = :shipId';
+        $stmt = $this->pdo->prepare($query);
+        $stmt->bindValue(':fleetId', $fleetShip->getFleet()->getId());
+        $stmt->bindValue('shipId', $fleetShip->getShip()->getId());
+        $stmt->execute();
+    }
+
+    public function fetchShipInFleetByIds($shipId, $fleetId): array
```

Comment:
I think you could make a helper method in Fleet that returns the ShipFleet when passing in the Ship